server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -137,6 +137,10 @@ public ActionRequestValidationException validate() {
         if (source != null && source.size() == 0 && scroll != null) {
             validationException = addValidationError("[size] cannot be [0] in a scroll context", validationException);
         }
+        if (source != null && source.rescores() != null && source.rescores().isEmpty() == false && scroll != null) {
+            DEPRECATION_LOGGER.deprecated("Using [rescore] for a scroll query is deprecated and will return a 400 error in future " +
```

Comment:
It would be nice to mention that the rescore will be ignored, as that may not be clear to the user only from the fact it's deprecated.